shards/aggregate.go

```diff
@@ -62,7 +62,12 @@ func (c *collectSender) Done() (_ *zoekt.SearchResult, ok bool) {
 	agg := c.aggregate
 	c.aggregate = nil
 
-	zoekt.SortFilesByScore(agg.Files)
+	if agg.HasRanks() {
+		zoekt.SortFilesByRank(agg.Files)
+	} else {
+		zoekt.SortFilesByScore(agg.Files)
+	}
```

Comment:
nit: HasRanks just looks at the files, make `SortFilesByScore` do this decision so you don't have to duplicate at each call site.

Reply:
done